transport/http2_client.go

```diff
@@ -114,14 +114,42 @@ func dial(fn func(context.Context, string) (net.Conn, error), ctx context.Contex
 	return dialContext(ctx, "tcp", addr)
 }
 
+func isTemporary(err error) bool {
```

Comment:
Is this function needed for you original issue? If not, please leave it to another fix.